pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRunningQueryResource.java

```diff
@@ -124,6 +125,9 @@ public String cancelQuery(
           .setSocketTimeout(timeoutMs).build();
 
       CloseableHttpClient client = HttpClients.custom().setConnectionManager(_httpConnMgr)
+          // The `setSSLSocketFactory` has to be set after `setConnectionManager` due to the bug described here:
+          // https://github.com/apache/pinot/issues/13431#issuecomment-2178323975
```

Comment:
Nit: Please don't mention the issue as a code comment. We can easily track this with the PR description.

Reply:
https://github.com/apache/pinot/pull/13477/commits/15108722de6ac77beb407be8bdffc8be549b1d91